components/shared/profile/edit/EditProfile.tsx

```diff
@@ -0,0 +1,229 @@
+    }));
+    console.log(data);
+    if (user?._id) {
+      try {
+        await updateUser({
+          updateData: {
+            fullname,
+            portfolio,
+            experiences: experiences?.map(
+              (experience: any) => experience.name as string
+            ),
+            technologies,
+            location,
+            learningGoals: goalArray,
+            availability: availability.map((date: any) => date),
+            socials,
+          },
+          path: pathname,
+        });
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  };
+
+  useEffect(() => {});
+  // TODO: substitute FormFields with regular tags
+  return (
+    <form
+      onSubmit={handleSubmit(onSubmit)}
+      className="space-y-6 flex flex-col px-[30px] w-full"
+    >
+      <div className="space-y-2">
+        <label htmlFor="fullname" className="block text-white-300">
+          Full Name
+        </label>
+        <Input
+          {...register("fullname")}
+          className="input bg-black-700 text-white-100"
+          placeholder="fullname"
+        />
+        {errors.fullname && (
+          <p className="text-red-500">{errors.fullname.message}</p>
+        )}
+      </div>
+      <div className="space-y-2">
+        <label htmlFor="email" className="block text-white-300">
+          Username
+        </label>
+        <Input
+          {...register("email")}
+          className="input bg-black-700 text-white-100"
+          placeholder="email"
+          readOnly
+        />
+      </div>
+      {/* Portfolio URL */}
+      <div className="space-y-2">
+        <label htmlFor="portfolio" className="block  text-white-300">
+          Portfolio URL
+        </label>
+        <Input
+          {...register("portfolio")}
+          className="bg-black-700 text-white-100"
+          placeholder="portfolio"
+        />
+      </div>
+      {user?.learningGoals &&
+        user.learningGoals.map((goal) => (
+          <div key={goal.name}>
+            <LearningGoal
+              completed={goal.completed}
+              name={goal.name}
+              onChange={() => removeGoal(...goal)}
+            />
+          </div>
+        ))}
+      {goalFields.map((field, index: number) => (
+        <div key={field.id} className="profile-goals-wrapper">
+          <Input
+            type="checkbox"
+            color="green"
+            className="size-9 order-1 bg-black-700 text-white-100"
+            id={`goals[${index}]CB`}
+            {...register(`goals[${index}].checked` as any)}
+          />
+          <Input
+            className="order-2"
+            id={`goals[${index}]`}
+            placeholder="Enter your goal"
+            {...register(`goals[${index}].name` as any)}
+          />
+          <Button
+            type="button"
+            className="text-white-100"
+            onClick={() => removeGoal(index)}
+          >
+            Remove
+          </Button>
+        </div>
+      ))}
```

Comment:
```suggestion
      {goalFields.map((field, index: number) => (
        <div key={field.id} className="profile-goals-wrapper">
          <Input
            type="checkbox"
            color="green"
            className="order-1 size-9 bg-black-700 text-white-100"
            id={`learningGoals[${index}]CB`}
            {...register(`learningGoals[${index}].completed` as any)}
          />
          <Input
            className="order-2"
            id={`learningGoals[${index}]`}
            placeholder="Enter your goal"
            {...register(`learningGoals[${index}].name` as any)}
          />
          <Button
            type="button"
            className="text-white-100"
            onClick={() => removeGoal(index)}
          >
            Remove
          </Button>
        </div>
      ))}
```

Learning goals is named "learningGoals" in your schema. so you have to change the name of all of them from the example I gave you. This should work.